crates/sui-graphql-client/src/lib.rs

```diff
@@ -145,6 +184,34 @@ impl Default for PaginationFilter<'_> {
     }
 }
 
+impl<T: Serialize> From<T> for NameValue {
+    fn from(value: T) -> Self {
+        NameValue(bcs::to_bytes(&value).unwrap())
+    }
+}
+
+impl From<BcsName> for NameValue {
+    fn from(value: BcsName) -> Self {
+        NameValue(value.0)
+    }
+}
+
+impl DynamicFieldOutput {
+    pub fn deserialize<T: DeserializeOwned>(
+        &self,
+        expected_type: &str,
+    ) -> Result<T, anyhow::Error> {
```

Comment:
This is a broader question than for this PR, but what are our thoughts on using `anyhow` to carry SDK errors? It gives us a lot of flexibility, but it makes the error types opaque for users (e.g. suppose I want to treat an "object not found" error differently from other errors, that's difficult).

cc @bmwill

Reply:
Ideally we don't use anyhow at all and instead have more detailed errors. anyhow doesn't impl Error itself so it doesn't really play very nicely when used in a library. On the far other end of the spectrum I'm very against the SuiError type of thing where we have a single bucket where there's a variant for everything. Errors should try to be focused and when they need to be more general we can utilize the "source" paradigm where we curry a Box<dyn Error> inside of our error that users can downcast to if they so desire.

Reply:
That's next on my agenda, to rework all the anyhow errors into explicit errors.